src/services/auth.service.ts

```diff
@@ -101,3 +108,104 @@ export const findOrCreateUser = async (
 
   return user
 }
+
+export const generateResetToken = async (
+  email: string
+): Promise<{ statusCode: number; message: string; token: string }> => {
+  try {
+    const profileRepository = dataSource.getRepository(Profile)
+    const profile = await profileRepository.findOne({
+      where: { primary_email: email },
+      select: ['password', 'uuid']
+    })
+
+    if (!profile) {
+      return {
+        statusCode: 401,
+        message: 'Invalid email or password',
+        token: ''
+      }
+    }
+    const token = jwt.sign({ userId: profile.uuid }, JWT_SECRET, {
+      expiresIn: '1h'
+    })
+    // Getting the email content
+    const content = getPasswordResetEmailContent(email, token)
+    // Sending the email with token
+    if (content) {
+      await sendResetPasswordEmail(email, content.subject, content.message)
+    }
+    return {
+      statusCode: 200,
+      message: 'Password reset link successfully sent to email',
+      token
+    }
+  } catch (error) {
+    console.error(
+      'Error executing Reset Password && Error sending password reset link',
+      error
+    )
+    return { statusCode: 500, message: 'Internal server error', token: '' }
+  }
+}
+
+const hashPassword = async (password: string): Promise<string> => {
+  return await bcrypt.hash(password, 10)
+}
+
+const saveProfile = async (
+  profile: Profile,
+  hashedPassword: string
+): Promise<void> => {
+  profile.password = hashedPassword
+  await dataSource.getRepository(Profile).save(profile)
+}
+
+export const resetPassword = async (
+  token: string,
+  newPassword: string
+): Promise<{ statusCode: number; message: string }> => {
+  if (!token || !newPassword) {
+    console.error('Error executing Reset Password: Missing parameters')
+    return { statusCode: 400, message: 'Missing parameters' }
+  }
+
+  let decoded
+  try {
+    decoded = jwt.verify(token, JWT_SECRET) as { userId: string }
+  } catch (error) {
+    console.error('Error executing Reset Password: Invalid token', error)
```

Comment:
Just throw the error. It will be catched by the try catch

Reply:
like this   ```
let decoded
  try {
    decoded = jwt.verify(token, JWT_SECRET) as { userId: string }
  } catch (error) {
    throw new Error('Invalid token')
  }
```